src/Symfony/Component/VarDumper/Caster/FFICaster.php

```diff
@@ -0,0 +1,232 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use FFI\CData;
+use FFI\CType;
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * Casts FFI extension classes to array representation.
+ *
+ * @author Nesmeyanov Kirill <nesk@xakep.ru>
+ *
+ * @final
+ */
+class FFICaster
+{
+    /**
+     * In case of "char*" contains a string, the length of which depends on
+     * some other parameter, then during the generation of the string it is
+     * possible to go beyond the allowable memory area.
+     *
+     * This restriction serves to ensure that during processing does not take
+     * up the entire allowable PHP memory limit.
+     */
+    private const MAX_STRING_LENGTH = 255;
+
+    /**
+     * List of FFI scalar types.
+     */
+    private const FFI_SCALAR_TYPES = [
+        CType::TYPE_FLOAT,
+        CType::TYPE_DOUBLE,
+        CType::TYPE_UINT8,
+        CType::TYPE_SINT8,
+        CType::TYPE_UINT16,
+        CType::TYPE_SINT16,
+        CType::TYPE_UINT32,
+        CType::TYPE_SINT32,
+        CType::TYPE_UINT64,
+        CType::TYPE_SINT64,
+        CType::TYPE_BOOL,
+        CType::TYPE_CHAR,
+    ];
+
+    public static function castCData(CData $data, array $args, Stub $stub): array
+    {
+        $type = \FFI::typeof($data);
+        $stub->class = self::getClassName($type, $data);
+
+        return match (true) {
+            self::isScalar($type) => self::castFFIScalar($data),
+            self::isEnum($type) => self::castFFIEnum($data),
+            self::isPointer($type) => self::castFFIPointer($stub, $type, $data),
+            self::isStruct($type) => self::castFFIStruct($type, $data),
+            self::isFunction($type) => self::castFFIFunction($stub, $type),
+            default => $args,
+        };
+    }
+
+    public static function castCType(CType $type, array $args, Stub $stub): array
+    {
+        $stub->class = self::getClassName($type, $type);
+
+        return match (true) {
+            self::isScalar($type) => self::castFFIScalar(),
+            self::isEnum($type) => self::castFFIEnum(),
+            self::isPointer($type) => self::castFFIPointer($stub, $type),
+            self::isStruct($type) => self::castFFIStruct($type),
+            self::isFunction($type) => self::castFFIFunction($stub, $type),
+            default => $args,
+        };
+    }
+
+    private static function isFunction(CType $type): bool
+    {
+        return CType::TYPE_FUNC === $type->getKind();
+    }
+
+    private static function isPointer(CType $type): bool
+    {
+        return CType::TYPE_POINTER === $type->getKind();
+    }
+
+    private static function isEnum(CType $type): bool
+    {
+        return CType::TYPE_ENUM === $type->getKind();
+    }
+
+    private static function isStruct(CType $type): bool
+    {
+        return CType::TYPE_STRUCT === $type->getKind();
```

Comment:
these methods look superfluous to me, better inline

Reply:
Here I was guided by uniformity in the chain (i mean `return match (true) { isX => xxx, isY => yyy }`) of checks. Doesn't that improve readability?

Reply:
I just thought that this uniformity is not so critical. So I fixed it